src/Microsoft.Health.Dicom.Functions/DurableFunctionsTimedCleanup.cs

```diff
@@ -0,0 +1,81 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Functions.Indexing.Configuration;
+
+namespace Microsoft.Health.Dicom.Functions
+{
+    public class DurableFunctionsTimedCleanup
+    {
+        private readonly ReindexErrorConfiguration _cleanupConfig;
+        private readonly Func<DateTime> _getUtcNow;
+
+        public DurableFunctionsTimedCleanup(IOptions<ReindexErrorConfiguration> cleanupOptions)
+            : this(cleanupOptions, () => DateTime.UtcNow)
+        { }
+
+        internal DurableFunctionsTimedCleanup(IOptions<ReindexErrorConfiguration> cleanupOptions, Func<DateTime> getDateTimeUtcNow)
+        {
+            EnsureArg.IsNotNull(cleanupOptions, nameof(cleanupOptions));
+            EnsureArg.IsNotNull(getDateTimeUtcNow, nameof(getDateTimeUtcNow));
+            _cleanupConfig = cleanupOptions.Value;
+            _getUtcNow = getDateTimeUtcNow;
+        }
+
+        [FunctionName("DurableFunctionsTimedCleanup")]
+        public async Task Run([TimerTrigger("*/15 * * * * *")] TimerInfo myTimer,
+            [DurableClient] IDurableOrchestrationClient client,
+            ILogger log,
+            CancellationToken hostCancellationToken)
+        {
+            List<string> orchestrationInstanceIdList = new List<string>();
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+
+            EnsureArg.IsNotNull(client, nameof(client));
+            EnsureArg.IsNotNull(myTimer, nameof(myTimer));
+
+            log.LogInformation("Durable Functions cleanup started at: " + myTimer.ToString());
+            string isPastDue = myTimer.IsPastDue ? "Current function invocation is later than scheduled." : "Function invocation is on schedule.";
+            log.LogInformation(isPastDue);
```

Comment:
nit: We can perhaps simply log a `Warning` if the timer is past due. Otherwise we don't log anything